lxd/storage/drivers/driver_zfs.go

```diff
@@ -354,12 +354,6 @@ func (d *zfs) Create() error {
 
 	revert.Add(func() { _ = d.Delete(nil) })
 
-	// Apply our default configuration.
-	err = d.ensureInitialDatasets(false)
```

Comment:
the bool argument to ensureInitialDatasets in Create and Mount differ though

Reply:
That's true.

A different approach might be to _not_ unmount the storage pool in the `zfs.Create` function. We could then remove the `pool.Mount` call in `storagePoolCreateLocal`. That would remove one call to `ensureInitialDatasets`. Let me check if our tests like that.